src/workshed/index.ts

```diff
@@ -0,0 +1,143 @@
+import { getWorkshed, haveEffect, Item, print, totalTurnsPlayed, use, visitUrl } from "kolmafia";
+import { $effect, $item, $items, AsdonMartin, clamp, DNALab, get, have, TrainSet } from "libram";
+import { dietCompleted } from "../diet";
+import { globalOptions } from "../config";
+import { potionSetupCompleted } from "../potions";
+import { estimatedTurns } from "../turns";
+import { getBestCycle, grabMedicine, offsetDefaultPieces } from "./utils";
+
+let _nextWorkshed: GarboWorkshed | null = null;
+let _currentWorkshed: GarboWorkshed | null = null;
+class GarboWorkshed {
+  workshed: Item;
+  done = () => true;
+  action = () => {
+    return;
+  };
+  constructor(workshed: Item, done?: () => boolean, action?: () => void) {
+    this.workshed = workshed;
+    if (done) this.done = done;
+    if (action) this.action = action;
+  }
+
+  static getGarboWorkshed(item: Item): GarboWorkshed {
+    return worksheds.find(({ workshed }) => workshed === item) ?? defaultWorkshed;
+  }
+
+  static currentWorkshed(): GarboWorkshed {
+    _currentWorkshed ??= GarboWorkshed.getGarboWorkshed(getWorkshed());
+    return _currentWorkshed ?? defaultWorkshed;
+  }
+
+  static nextWorkshed(): GarboWorkshed {
+    _nextWorkshed ??= GarboWorkshed.getGarboWorkshed(globalOptions.workshed);
+    return _nextWorkshed ?? defaultWorkshed;
+  }
+}
+const defaultWorkshed = new GarboWorkshed($item`none`);
+const estimatedTurnsTomorrow = 400 + clamp((get("valueOfAdventure") - 4000) / 8, 0, 600);
```

Comment:
where... where did this number come from? i did some quick math and i think it "works" but i do fear this is a little bit of a "this works now but won't work later" type thing as different turngen sources drop or get nerfed. any appetite to make something like a "maxGarboTurnsPlayed" preference that garbo updates if you spent more turns in a given garbo session than the preference? that way it isn't VoA based but usage based, and using the highest # of turns the user has ever had. idk. just a thought.

Reply:
This one is indeed magic numbers, but the `clamp` makes this harder to figure out. I would like to have a fuller discussion about this in the LASS channel but I'm yet to see a better pitch